File: app/src/main/java/com/kuplay/kuplay/base/BaseView.java
package com.kuplay.kuplay.base;


import com.kuplay.kuplay.util.ToastManager;

/**
 * Created by "iqos_jay@outlook.com" on 2018/6/22.
 * BaseView
 */
public interface BaseView {
    /**
     * Show the toast tip.
     *
     * @param resId the message's id of what would you like to tip.
     */
    void onToast(int resId);

    /**
     * Show the toast tip
     *
     * @param msg the message of what would you like to tip.
     */
    void onToast(String msg);

    /**
     * Show the toast tip
     *
     * @param msg      the message's id of what would you like to tip.
     * @param duration the message last time.
     */
    void onToast(int msg, @ToastManager.Duration int duration);

    /**
     * Show the toast tip
     *
     * @param msg      the message of what would you like to tip
     * @param duration the message last time.
     */
    void onToast(String msg, @ToastManager.Duration int duration);

    /**
     * Hide the content being prompted.
     */
    void onHideToast();
}

File: pi_framework/src/main/java/com/kuplay/pi_framework/base/BaseView.kt
package com.kuplay.pi_framework.base

import com.kuplay.pi_framework.Util.ToastManager

interface BaseView {
    /**
     * Show the toast tip.
     *
     * @param resId the message's id of what would you like to tip.
     */
    fun onToast(resId: Int)

    /**
     * Show the toast tip
     *
     * @param msg the message of what would you like to tip.
     */
    fun onToast(msg: String)

    /**
     * Show the toast tip
     *
     * @param msg      the message's id of what would you like to tip.
     * @param duration the message last time.
     */
    fun onToast(msg: Int, @ToastManager.Duration duration: Int)

    /**
     * Show the toast tip
     *
     * @param msg      the message of what would you like to tip
     * @param duration the message last time.
     */
    fun onToast(msg: String, @ToastManager.Duration duration: Int)

    /**
     * Hide the content being prompted.
     */
    fun onHideToast()
}